web/modules/custom/dpl_event/src/Form/SettingsForm.php

```diff
@@ -98,14 +100,66 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       15552000,
     ];
     $period = array_map([$this->dateFormatter, 'formatInterval'], array_combine($period, $period));
+
+    $form['unpublish']['unpublish_enable'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Unpublish eventinstances when they have occured (recommended)', [], ['context' => 'DPL event']),
+      '#default_value' => $config->get('unpublish_enable'),
+    ];
+
+    $form['unpublish']['unpublish_disable_warning'] = [
+      '#type' => 'container',
+      // js-form-wrapper is important - otherwise, Drupal states will not work.
+      '#prefix' => '<div class="dpl-form-warning js-form-wrapper">',
+      '#markup' => $this->t('Notice - if you do not choose that eventinstances get unpublished, they may show up in automatic and manual lists, across the site.', [], ['context' => 'DPL event']),
+      '#suffix' => '</div>',
+      '#states' => [
+        'visible' => [
+          ':input[name="unpublish_enable"]' => ['checked' => FALSE],
+        ],
+      ],
+    ];
+
     $form['unpublish']['unpublish_schedule'] = [
       '#type' => 'select',
-      '#title' => $this->t('Schedule', [], ['context' => "DPL event"]),
+      '#title' => $this->t('How much time should pass after an eventinstance has occurred before it should be unpublished?', [], ['context' => "DPL event"]),
       '#default_value' => $config->get('unpublish_schedule'),
       '#options' => $period,
-      '#empty_option' => $this->t('Automatic unpublication disabled', [], ['context' => "DPL event"]),
-      '#empty_value' => 0,
-      '#description' => $this->t('How much time should pass after an event has occurred before it should be unpublished automatically.', [], ['context' => "DPL event"]),
+      '#states' => [
+        'visible' => [
+          ':input[name="unpublish_enable"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['unpublish']['unpublish_series_enable'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Unpublish the series when all instances have occurred (not recommended)', [], ['context' => "DPL event"]),
+      '#default_value' => $config->get('unpublish_series_enable'),
+      // Only display the field when unpublish schedule has a non-0 value.
+      '#states' => [
+        'visible' => [
+          ':input[name="unpublish_enable"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['unpublish']['unpublish_series_enable_warning'] = [
+      '#type' => 'container',
+      // js-form-wrapper is important - otherwise, Drupal states will not work.
+      '#prefix' => '<div class="dpl-form-warning js-form-wrapper">',
+      '#markup' => $this->t('Notice - if series get unpublished, old instance links will no longer work. If you however keep the series published, expired instances will redirect to the associated series.', [], ['context' => 'DPL event']),
```

Comment:
When I think about it I wonder why we ended up with not recommending unpublishing series when the last instance has been unpublished.

If the intention is that lists should not contain irrelevant content then keeping series which only contain expired instances seem to be counterproductive. Perhaps this is what you agreed on with the business?

Reply:
Yes, this is what we agreed to with Stefan/the business:

For historical purposes, some libraries want to be able to display what has happened in the past.